src/pages/Option/Index.js

```diff
@@ -37,9 +43,73 @@ const StyledDrawer = styled(Drawer)`
     color: #b5b5b6;
   }
 `
+   
+function safeDiv(a, b) {
+    return b==0 ? 0 : a.div(b).toNumber();
+}
+
+export function getPosition(rawPositionData,symbolData) {
+  if (!rawPositionData || !symbolData) {
+    return
+  }
+  let positionQuery = []
+  for(let i = 0; i < rawPositionData.positions.length; i++){
+    const temp = rawPositionData.positions[i]
+    const markPrice = symbolData.find(obj => {
+      return obj.address === temp[0]
+    }).markPrice
+    const initialMarginRatio = symbolData.find(obj => {
+      return obj.address === temp[0]
+    }).initialMarginRatio
+    const indexPrice = symbolData.find(obj => {
+      return obj.address === temp[0]
+    }).indexPrice
+    const marginUsage = Math.abs(temp[2].toNumber() * indexPrice) * initialMarginRatio
```

Comment:
the `.toNumber()` would fail on my end because `temp[2]`, volume, is a bigNumber of 18 decimals. When user has opened a position of 1 BTC size, this value would be 1e18 in BigNumber, and overflows the conversion (check attached image). 

a correct way to parse it is `const volume = ethers.utils.formatUnits(temp[2], 18)`, and use `volume` anywhere below this line (no recomputation).

[**Tips**](https://docs.ethers.io/v5/api/utils/display-logic/#unit-conversion): `ethers.utils.formatUnits` converts BigNumber to string, `ethers.utils.parseUnits` converts string to BigNumber.

![1669188676697](https://user-images.githubusercontent.com/50497617/203496106-04c6cee0-2973-4479-96c1-d7490f7da0b2.jpg)